### PR TITLE
stable diffusion image variation/InstructPix2Pix/StableDiffusionXLImg2ImgPipeline pipeline

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sywangyi** [comment]
[image_to_image_generation.py.txt](https://github.com/huggingface/optimum-habana/files/15451492/image_to_image_generation.py.txt)
benchmark result with A100
stabilityai/stable-diffusion-xl-refiner-1.0:

command in A100
python image_to_image_generation.py \
    --model_name_or_path "stabilityai/stable-diffusion-xl-refiner-1.0" \
    --src_image_path "https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg" \
    --prompts "turn him into cyborg" \
    --num_images_per_prompt 1 \
    --cuda \
    --bf16 \
    --guidance_scale 7.5 \
    --num_inference_steps 10 \
    --image_save_dir /tmp/stable_diffusion_images

command in Gaudi2
python image_to_image_generation.py \
    --model_name_or_path "stabilityai/stable-diffusion-xl-refiner-1.0" \
    --src_image_path "https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg" \
    --prompts "turn him into cyborg" \
    --num_images_per_prompt 1 \
    --guidance_scale 7.5 \
    --num_inference_steps 10 \
    --bf16 \
    --use_habana \
    --use_hpu_graphs \
    --gaudi_config Habana/stable-diffusion \
    --image_save_dir /tmp/stable_diffusion_images

perf result A100 233ms Gaudi2 212ms

**Comment by sywangyi** [comment]
**"lambdalabs/sd-image-variations-diffusers"**
python image_to_image_generation.py \
    --model_name_or_path "lambdalabs/sd-image-variations-diffusers" \
    --src_image_path "https://github.com/SHI-Labs/Versatile-Diffusion/blob/master/assets/demo/reg_example/ghibli.jpg?raw=true" \
    --num_images_per_prompt 1 \
    --image_save_dir /tmp/stable_diffusion_images \
    --guidance_scale 3 \
    --bf16 \
    --cuda

Gaudi2 command
python image_to_image_generation.py \
    --model_name_or_path "lambdalabs/sd-image-variations-diffusers" \
    --src_image_path "https://github.com/SHI-Labs/Versatile-Diffusion/blob/master/assets/demo/reg_example/ghibli.jpg?raw=true" \
    --num_images_per_prompt 1 \
    --image_save_dir /tmp/stable_diffusion_images \
    --guidance_scale 3 \
    --bf16 \
    --use_habana \
    --use_hpu_graphs \
    --gaudi_config Habana/stable-diffusion
**perf result A100 1.593s Gaudi2 1.205s**

**Comment by sywangyi** [comment]
**timbrooks/instruct-pix2pix:**
command in A100
python image_to_image_generation.py \
    --model_name_or_path "timbrooks/instruct-pix2pix" \
    --src_image_path "https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg" \
    --prompts "turn him into cyborg" \
    --num_images_per_prompt 1 \
    --batch_size 1 \
    --guidance_scale 7.5 \
    --image_guidance_scale 1 \
    --num_inference_steps 10 \
    --image_save_dir /tmp/stable_diffusion_images \
    --bf16 --cuda

command in Gaudi2:
python image_to_image_generation.py \
    --model_name_or_path "timbrooks/instruct-pix2pix" \
    --src_image_path "https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg" \
    --prompts "turn him into cyborg"\
    --num_images_per_prompt 1 \
    --batch_size 1 \
    --guidance_scale 7.5 \
    --image_guidance_scale 1 \
    --num_inference_steps 10 \
    --image_save_dir /tmp/stable_diffusion_images \
    --use_habana \
    --use_hpu_graphs \
    --gaudi_config Habana/stable-diffusion \
    --bf16

**perf result. A100 468ms Gaudi2 398ms**

File: optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py [hunk 1]
```diff
@@ -0,0 +1,790 @@
...
+                ).to(device=device, dtype=latents.dtype)
+
+            self._num_timesteps = len(timesteps)
+
+            # 8.3 Denoising loop
+            throughput_warmup_steps = kwargs.get("throughput_warmup_steps", 3)
+            for j in self.progress_bar(range(num_batches)):
+                # The throughput is calculated from the 3rd iteration
+                # because compilation occurs in the first two iterations
+                if j == throughput_warmup_steps:
```
Comment (dsocek) [inline optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py:552]: Please use updated logic for warmup here. See https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L498
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L507
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L601

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_image_variation.py [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+
+            # 8. Denoising loop
+            t0 = time.time()
+            t1 = t0
+            throughput_warmup_steps = kwargs.get("throughput_warmup_steps", 3)
+            num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+            for j in self.progress_bar(range(num_batches)):
+                # The throughput is calculated from the 3rd iteration
+                # because compilation occurs in the first two iterations
+                if j == throughput_warmup_steps:
```
Comment (dsocek) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_image_variation.py:327]: Please use updated logic for warmup here. See https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L498
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L507
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L601

File: optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_instruct_pix2pix.py [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+
+            # 10. Denoising loop
+            t0 = time.time()
+            t1 = t0
+            throughput_warmup_steps = kwargs.get("throughput_warmup_steps", 3)
+            num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+            for j in self.progress_bar(range(num_batches)):
+                # The throughput is calculated from the 3rd iteration
+                # because compilation occurs in the first two iterations
+                if j == throughput_warmup_steps:
```
Comment (dsocek) [inline optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_instruct_pix2pix.py:410]: Please use updated logic for warmup here. See https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L498
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L507
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L601

**Review by dsocek — COMMENTED** [review]
Very clean PR.   Please update warmup logic steps (see comments).

**Review by regisss — APPROVED** [review]

**Pull request merged**